contracts/market-wrapper/FractionalMarketWrapper.sol

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// ============ External Imports ============
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {TokenVault} from "../external/fractional/ERC721TokenVault.sol";
+import {ERC721VaultFactory} from "../external/fractional/ERC721VaultFactory.sol";
+import {Settings} from "../external/fractional/Settings.sol";
+
+// ============ Internal Imports ============
+import {IMarketWrapper} from "./IMarketWrapper.sol";
+
+/**
+ * @title FractionalMarketWrapper
+ * @author vick.eth, Jacob Frantz
+ * @notice MarketWrapper contract implementing IMarketWrapper interface
+ * according to the logic of Fractional's TokenVaults
+ * Original Fractional TokenVault code: https://github.com/fractional-company/contracts/blob/master/src/ERC721TokenVault.sol
+ */
+
+contract FractionalMarketWrapper is IMarketWrapper {
+    using SafeMath for uint256;
+
+    // ============ Internal Immutables ============
+
+    ERC721VaultFactory internal immutable vaultFactory;
+    Settings internal immutable settings;
+
+    // ======== Constructor =========
+
+    constructor(address _fractionalVaultFactory) {
+        vaultFactory = ERC721VaultFactory(_fractionalVaultFactory);
+        settings = Settings(
+            ERC721VaultFactory(_fractionalVaultFactory).settings()
+        );
+    }
+
+// ======== External Functions =========
+
+    /**
+     * @notice Given the auctionId, nftContract, and tokenId, check that:
+     * 1. the auction ID matches the token
+     * referred to by tokenId + nftContract
+     * 2. the auctionId refers to an *ACTIVE* auction
+     * (e.g. an auction that will accept bids)
+     * within this market contract
+     * 3. any additional validation to ensure that
+     * a PartyBid can bid on this auction
+     * (ex: if the market allows arbitrary bidding currencies,
+     * check that the auction currency is ETH)
+     * Note: This function probably should have been named "isValidAuction"
+     * @dev Called in PartyBid.sol in `initialize` at line 174
+     * @return TRUE if the auction is valid
+     */
+    function auctionIdMatchesToken(
+        uint256 auctionId,
+        address nftContract,
+        uint256 tokenId
+    ) external override view returns (bool) {
+        TokenVault auction = TokenVault(vaultFactory.vaults(auctionId));
```

Comment:
i like the workaround of querying the token vault address from the factory using the `auctionId` - smart.